controllers/authorization/authorization_controller_test.go

```diff
@@ -1,15 +1,172 @@
+
+			Expect(createdAuthConfig).To(HaveAuthenticationMethod("anonymous-access"))
+			Expect(createdAuthConfig).NotTo(HaveAuthenticationMethod("kubernetes-user"))
+			Expect(createdAuthConfig).NotTo(HaveKubernetesTokenReview())
+
+			return nil
+		}, 10*time.Second, 2*time.Second).Should(Succeed())
+	})
+
+	It("should create a non-anonymous AuthConfig resource when annotation is specified", func(ctx context.Context) {
+		if createdComponent.GetAnnotations() == nil {
+			createdComponent.SetAnnotations(map[string]string{})
+		}
+
+		annotations := createdComponent.GetAnnotations()
+		annotations[metadata.Annotations.AuthEnabled] = "true"
+		createdComponent.SetAnnotations(annotations)
+		Expect(envTest.Client.Update(ctx, createdComponent)).To(Succeed())
+
+		Eventually(func() error {
+			createdAuthConfig := &authorinov1beta2.AuthConfig{}
+			err := envTest.Client.Get(ctx, types.NamespacedName{
+				Name:      resourceName,
+				Namespace: testNamespaceName,
+			}, createdAuthConfig)
+
+			if err != nil {
+				return err
+			}
+
+			Expect(createdAuthConfig).To(HaveHosts("example.com"))
+			Expect(createdAuthConfig.Labels).To(HaveKeyWithValue("security.opendatahub.io/authorization-group", "default"))
+
+			Expect(createdAuthConfig).To(HaveAuthenticationMethod("kubernetes-user"))
+			Expect(createdAuthConfig).NotTo(HaveAuthenticationMethod("anonymous-access"))
+			Expect(createdAuthConfig).To(HaveKubernetesTokenReview())
+
+			return nil
+		}, 10*time.Second, 2*time.Second).Should(Succeed())
+	})
+
+	It("should create an AuthorizationPolicy when a Component is created", func(ctx context.Context) {
+		Eventually(func() error {
+			createdAuthPolicy := &istiosecurityv1beta1.AuthorizationPolicy{}
+			err := envTest.Client.Get(ctx, types.NamespacedName{
+				Name:      resourceName,
+				Namespace: testNamespaceName,
+			}, createdAuthPolicy)
+
+			if err != nil {
+				return err
+			}
+
+			Expect(createdAuthPolicy.Spec.GetAction()).To(Equal(v1beta1.AuthorizationPolicy_CUSTOM))
+
+			return nil
+		}, 10*time.Second, 2*time.Second).Should(Succeed())
+	})
+
+	It("should create a PeerAuthentication when a Component is created", func(ctx context.Context) {
+		Eventually(func() error {
+			createdPeerAuth := &istiosecurityv1beta1.PeerAuthentication{}
+			err := envTest.Client.Get(ctx, types.NamespacedName{
+				Name:      resourceName,
+				Namespace: testNamespaceName,
+			}, createdPeerAuth)
+
+			if err != nil {
+				return err
+			}
+
+			Expect(createdPeerAuth.Spec.GetMtls().GetMode()).To(Equal(v1beta1.PeerAuthentication_MutualTLS_PERMISSIVE))
+
+			return nil
+		}, 10*time.Second, 2*time.Second).Should(Succeed())
+	})
+
+	// TODO: fill out stubs once owner-name labels are propagated to auth resources
+	PIt("should have ownerReference on all created auth resources", func(ctx context.Context) {
+		// get all three resources by owner name label
+
+		// use matchers.owner.go to ensure that correct ownerReference is set on all of them
+	})
```

Comment:
This could be a follow-up PR where we can enhance reconcilers by adding `metadata.WithOwnerLabels(target)` to each reconciled resource.